aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -311,16 +356,32 @@ private void completeUploads() {
           .bucket(location.bucket())
           .key(location.key());
 
+      if (isEtagCheckEnabled) {
+        requestBuilder.contentMD5(BinaryUtils.toBase64(completeMessageDigest.digest()));
+      }
+
       S3RequestUtil.configureEncryption(awsProperties, requestBuilder);
       S3RequestUtil.configurePermission(awsProperties, requestBuilder);
 
-      s3.putObject(requestBuilder.build(), RequestBody.fromInputStream(contentStream, contentLength));
+      PutObjectRequest putObjectRequest = requestBuilder.build();
+      try {
+        s3.putObject(putObjectRequest, RequestBody.fromInputStream(contentStream, contentLength));
+      } catch (UncheckedIOException uncheckedIOException) {
+        checkProtocolException(uncheckedIOException, putObjectRequest.toString());
+      }
     } else {
       uploadParts();
       completeMultiPartUpload();
     }
   }
 
+  private static void checkProtocolException(UncheckedIOException uncheckedIOException, String requestString) {
+    if (uncheckedIOException.getCause() instanceof ProtocolException) {
+      LOG.error("S3 Request Failure: {}", requestString);
+      throw uncheckedIOException;
```

Comment:
This only throws protocol exceptions? What is being suppressed by this? Is there a guarantee that this is safe?

Reply:
Oh this is a mistake. Thanks for catching this @rdblue . Will fix it shortly